core/common/src/main/java/alluxio/underfs/UnderFileSystemFactoryRegistry.java

```diff
@@ -129,13 +145,55 @@ public static UnderFileSystemFactory find(String path) {
         eligibleFactories.add(factory);
       }
     }
+    if (eligibleFactories.isEmpty()) {
```

Comment:
that is by design. do you feel the semantics should be always scan instead?. the present method just avoids instantiating factories from extensions if not required